src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_SEGMENT_LENGTH = 100;
+    public static final int FIGURE_QUANTITY = 5;
+    private static final int DEFAULT_NUMBER = 10;
+    private final Random figureNumber = new Random();
+
+    public Figure getRandomFigure() {
+        switch (figureNumber.nextInt(FIGURE_QUANTITY)) {
+            case 0:
+                return new Circle(new ColorSupplier().getRandomColor(), new Random()
+                        .nextInt(MAX_SEGMENT_LENGTH));
+            case 1:
+                return new Rectangle(new ColorSupplier().getRandomColor(), new Random()
```

Comment:
```suggestion
                return new Rectangle(colorSuplier.getRandomColor(), new Random()
```
ColorSupplier also should  make class-level variable

Reply:
Fixed.
Just a question.
I understand why do I need to create object for Random (cause it prevents the multiple creation of new object during a 'Figure' object creation.
But in ColorSupplier case we create only one object anyway. So I should to create class-level variable 'colorSuplier' not because of performance issues, but for better code readability and style.
Am I right?